### PR TITLE
Fixed PHPStan error with Mage_Catalog_Model_Resource_Product_Option_Collection

### PR DESCRIPTION
**Pull request by fballiano**
Discovered in https://github.com/OpenMage/magento-lts/pull/2562 by @elidrissidev, it shouldn't be caused with that PR but anyway this PR should fix it.

Let's see if all lights are gree.

### CODE REVIEW

File: app/code/core/Mage/Catalog/Model/Resource/Product/Option/Collection.php [hunk 1]
```diff
@@ -149,8 +149,9 @@ public function addValuesToResult($storeId = null)
             $optionIds[] = $option->getId();
         }
         if (!empty($optionIds)) {
-            $values = Mage::getModel('catalog/product_option_value')
-                ->getCollection()
+            /** @var Mage_Catalog_Model_Resource_Product_Option_Value_Collection $values */
+            $values = Mage::getModel('catalog/product_option_value')->getCollection();
+            $values
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Model/Resource/Product/Option/Collection.php:154]: No need to instantiate model but may be not BC. 
```suggestion   
            $collection = Mage::getResourceModel('catalog/product_option_value_collection');
            $collection
```

Comment (fballiano) [reply app/code/core/Mage/Catalog/Model/Resource/Product/Option/Collection.php:154]: I don't know, probably nobody did an override on that class and the getcollection method but this type of code is all over the code, I wouldn't change it in this specific PR

**Review by elidrissidev — APPROVED** [review]

**Comment by fballiano** [comment]
I found where it originated and fixed it reverting previous commits

**Review by elidrissidev — APPROVED** [review]

**Comment by ADDISON74** [comment]
All Green.

**Comment by fballiano** [comment]
yeah, i was waiting for the checks to end :-)

**Pull request merged**